src/components/UpcomingBirthdays/UpcomingBirthdays.js

```diff
@@ -1,9 +1,40 @@
 import './UpcomingBirthdays.css';
+import { useSelector } from 'react-redux';
 
 const UpcomingBirthdays = () => {
+  const friends = useSelector(state => state.friends);
+  const upcomingBirthdays = friends.filter(friend => {
+    const today = new Date();
+    const monthFromToday = new Date(today.getTime() + 30*24*60*60*1000);
+    const currentDate = today.toISOString().split('T')[0];
+    const currentMonthFromToday = monthFromToday.toISOString().split('T')[0];
+    return friend.birthday > currentDate && friend.birthday < currentMonthFromToday
+  });
+  const sortedBirthdays = upcomingBirthdays.sort((a, b) => new Date(a.birthday) - new Date(b.birthday))
+  const relevantBirthdays = sortedBirthdays.map(friend => {
+    return (
+      <article key={friend.id}>
+        <ul className='name-area'>
+          {friend.gift === 'purchased ' &&
+          <div className='check-area'>
+            <i className="fas fa-check">{friend.name}</i>
+          </div>}
+          {friend.gift === 'pending ' && <li>{friend.name}</li>}
+        </ul>
+          <ul>
+            <li>{friend.birthday.split('2021-')[1]}</li>
+            <li>{friend.gift}</li>
```

Comment:
I like the inclusion of the gift status to help make it clear to the user who they need to still buy a present for, but I think it might create a better UX if we make the terms more straightforward. Purchased could be "Gift purchased" and pending could be "Gift shipped" or something like that.

Reply:
Yeah good call maybe we can talk to the backend for more clarification.